manim/mobject/text/text_mobject.py

```diff
@@ -197,15 +179,24 @@ def _gen_chars(self, lines_str_list):
         char_index_counter = 0
         chars = self.get_group_class()()
         for line_no in range(lines_str_list.__len__()):
+            line_str = lines_str_list[line_no]
+            # Count all the characters in line_str
+            # Spaces may or may not count as characters
+            char_count = line_str.__len__()
+            if not self.consider_spaces_as_chars:
+                for blank in (" ", "\n", "\t"):
+                    char_count -= line_str.count(blank)
```

Comment:
Instead of hardcoding these particular spaces, why not use the inbuilt `isspace` method of strings? It also detects `\f` and `\r`, which should make this even more robust.

```suggestion
                for char in line_str:
                    if char.isspace():
                        char_count -= 1
```

Reply:
In case you prefer a more concise version:

```suggestion
                char_count -= len([char for char in line_str if char.isspace()])
```

Reply:
Ooh, I didn't know there was an `isspace` method for strings! That's new for me 😮 
(also I didn't know about `\f` and `\r`)
I like the first suggestion, because it's more readable.
However, before I apply it, should I change `Text._gen_chars` as well? Because:

```python
    def _gen_chars(self):
        chars = self.get_group_class()()
        submobjects_char_index = 0
        for char_index in range(self.text.__len__()):
            if self.text[char_index] in (" ", "\t", "\n"):
                space = Dot(radius=0, fill_opacity=0, stroke_opacity=0)
                if char_index == 0:
                    space.move_to(self.submobjects[submobjects_char_index].get_center())
                else:
                    space.move_to(
                        self.submobjects[submobjects_char_index - 1].get_center(),
                    )
                chars.add(space)
            else:
                chars.add(self.submobjects[submobjects_char_index])
                submobjects_char_index += 1
        return chars
```

Besides this `self.text.__len__()` being used in the for loop (and some others in `Paragraph` lol), there's this condition `if self.text[char_index] in (" ", "\t", "\n"):`. I should probably also change this to `if self.text[char_index].isspace():`, right?